crates/ironrdp-web/src/session.rs

```diff
@@ -492,18 +477,122 @@ impl Session {
                     }
                     ActiveStageOutput::PointerDefault => {
                         let _ret = self
-                            .show_pointer_callback
-                            .call0(&self.show_pointer_callback_context)
-                            .map_err(|e| anyhow::Error::msg(format!("show pointer callback failed: {e:?}")))?;
+                            .set_cursor_style_callback
+                            .call3(
+                                &self.set_cursor_style_callback_context,
+                                &JsValue::from_str("default"),
+                                &JsValue::UNDEFINED,
+                                &JsValue::UNDEFINED,
+                            )
+                            .map_err(|e| anyhow::Error::msg(format!("set cursor style callback failed: {e:?}")))?;
                     }
                     ActiveStageOutput::PointerHidden => {
                         let _ret = self
-                            .hide_pointer_callback
-                            .call0(&self.hide_pointer_callback_context)
-                            .map_err(|e| anyhow::Error::msg(format!("hide pointer callback failed: {e:?}")))?;
+                            .set_cursor_style_callback
+                            .call3(
+                                &self.set_cursor_style_callback_context,
+                                &JsValue::from_str("none"),
+                                &JsValue::UNDEFINED,
+                                &JsValue::UNDEFINED,
+                            )
+                            .map_err(|e| anyhow::Error::msg(format!("set cursor style callback failed: {e:?}")))?;
                     }
                     ActiveStageOutput::PointerPosition { .. } => {
-                        // Not applicable for web
+                        // Not applicable for web.
+                    }
+                    ActiveStageOutput::NativePointerUpdate(pointer) => {
+                        // Maximum allowed cursor size for browsers is 32x32, because bigger sizes
+                        // will cause the following issues:
+                        // - 128x128 and bigger cursors are not supported in browsers.
+                        // - 32x32 and bigger cursors will default to the system cursor if their
```

Comment:
Nitpick: "cursors bigger than 128x128" and "cursors bigger than 32x32" would be a more correct wording, because 128x128 and 32x32 are actually _excluded_ (32x32 cursors are supported).